### DISCUSSION TITLE
JointJS+ FreeTransform resize for Node with multiple embedded childre

### DISCUSSION THREAD

**Discussion by ROYopedia** [clientIO/joint#2279] [Q&A] [answered]
### Introduction

Problem Statement:
I am using JointJS+ FreeTransform for resizing my custom element using the given example in jointjs docs.
paper.on('cell:pointerup', function(cellView) {
    // We don't want to transform links.
    if (cellView.model instanceof joint.dia.Link) return;

    var freeTransform = new joint.ui.FreeTransform({ cellView: cellView });
    freeTransform.render();
});
Now, 
The parent element has multiple children embedded inside it. While resizing the parent element the inside embedded childrens are going outside the parent element. 

Solution Tried:
using resize:stop events I tried to use fitEmbeds.
```js
freeTransform.on({
    "resize:stop": function () {
       cellView.model.fitEmbeds(
         {
         padding: {
           left: NODE_PADDING_HORIZONTAL,
           right: NODE_PADDING_HORIZONTAL,
           top: NODE_PADDING_VERTICAL + cellView.model.HEADER_HEIGHT + 50,
           bottom: NODE_PADDING_VERTICAL,
        },
       } );
    },
  });
 ```
Outcome is not resizable if we use fitEmbeds.

Solution Needed:
After parent resize all the children embedded inside the parent should not go outside of the parent.

Any suggestions/idea/or small demo would be appriciable.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment]
In version `3.7` there is a new `expandOnly` (and `shrinkOnly`) option which can help you with this.
Note that `fitEmbeds` has been renamed to `fitToChildren` (`fitEmbeds` will be removed in `v4`).
```js
freeTransform.on({
    'resize:stop': () => {
        element.fitToChildren({
            expandOnly: true,
            padding: 20
        });
    },
});
```

**Reply by ROYopedia** [reply]
I understand, This would fit the children but not resize the children along with the parent. Is there any use cases you guys came across?
In actual resize scenario for leading editors - grouped/embedded items are transforming/resizing too when we resize the grouped resize.

**Reply by kumilingus** [reply, marked as answer]
I misunderstood your question then. Resizing child elements is not always what you want. They may have a constant size, while containers do not.

If you want to resize/rotate an element along with its children you can use the `ui.Selection`. See this [example](https://codepen.io/jointjs/pen/xxQMopw/a5d4b3f532c574a08583fd09bba4fae8).
